src/integrations/default.ts

```diff
@@ -0,0 +1,28 @@
+import type { Integration } from '@sentry/types';
+
+import type { CapacitorOptions } from '../options';
+import { deviceContextIntegration } from './devicecontext';
+import { eventOriginIntegration } from './eventorigin';
+import { nativeReleaseIntegration } from './release';
+import { capacitorRewriteFramesIntegration } from './rewriteframes';
+import { sdkInfoIntegration } from './sdkinfo';
+
+/**
+ * Returns the default Capacitor integrations based on the current environment.
+ */
+export function getDefaultIntegrations(
+  options: CapacitorOptions,
+): Integration[] {
+  const integrations: Integration[] = [];
+
+  integrations.push(capacitorRewriteFramesIntegration);
+  integrations.push(nativeReleaseIntegration);
+  integrations.push(eventOriginIntegration);
+  integrations.push(sdkInfoIntegration);
+
+  if (options.enableNative) {
+    integrations.push(deviceContextIntegration);
```

Comment:
The integration array must hold the integration objects not the creator functions.

This way sadly TS doesn't error but none of the integrations would be called.

Reply:
Nice catch!